Liquidsoap 2.0 migration

gammaw opened the pull request:
This is a major update from Liquidsoap 1.4 to 2.X containing not backwards-compatible syntactical changes. Update of Azuracast will be necessary. 

**Main feature changes** (rationale for all: cleaner script and easier implementation): 
- Standard transition sequence for all track change: one station ID + one jingle in a sequence; 
   - So far: it varied based on playlist (e.g., OFF AIR vs scheduled shows); 
- Standard fading function for all track change: fade out + fade in; 
   - So far: it varied based on playlist (e.g., no fade in for scheduled shows); 
- Tracks from OFF AIR and ambient playlists will not be skipped (when interrupted); 
   - So far: current track was skipped when interrupted by a scheduled show; 

Note: 
- `once` operator is still in-place as there are issues with `delay` (see https://github.com/savonet/liquidsoap/discussions/2791). 
- The main file to review is `azuracast/liquidsoap_custom.liq`, the others are mainly legacy deletions. 

Related issue: https://github.com/lahmacunradio/liquidsoap/issues/422

Please review asap. Never mind that branch cannot be merged automatically, it will need to be a manual overwrite.

tuz666 reviewed (changes requested):
I checked the part what I know from liquidsoap_custom.liq, found 1 bigger and some minor stuff.

gammaw commented on azuracast/liquidsoap_custom.liq:
```diff
@@ -1,488 +1,512 @@
 
-playlist_queer = playlist(id="playlist_queer",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/lahmacun_radio/playlists/playlist_queer.m3u")
-playlist_queer = audio_to_stereo(id="stereo_playlist_queer", playlist_queer)
+playlist_tudtad = playlist(id="playlist_tudtad",mime_type="audio/x-mpegurl",mode="randomize",reload_mode="watch","/var/azuracast/stations/lahmacun_radio/playlists/playlist_tudtad.m3u")
```
Yes, thanks!

gammaw commented on azuracast/liquidsoap_custom.liq:
```diff
@@ -1,488 +1,512 @@
+    ({1w and 19h30m-20h58m}, once(playlist_kolegak)),
+    ({1w and 21h-21h58m}, once(playlist_urmajmok)),      
+    ({1w and 22h-23h58m}, once(playlist_infinite_scroll)),      
+    ({1w and 13h30m-23h30m}, playlist_off_air_ambient),                        #ambient: Who Say - Infinite
```
Yeah we can do that. It was like this because of the longer gap after tansverszia, but I changed it now.

gammaw commented on azuracast/liquidsoap_custom.liq:
```diff
@@ -1,488 +1,512 @@
+    ({3w and 20h-20h58m}, once(playlist_boombap)),
+    ({3w and 21h-21h58m}, once(playlist_eastern_daze)),
+    ({3w and 23h-23h58m}, once(playlist_linear_systems_with_gestalt)),
+    ({3w and 15h30m-23h30m}, playlist_off_air_ambient),                     #ambient: EH - Gestalt
```
Same like above, but here the gap is really long so I'll keep it as it is for now.

gammaw commented on azuracast/liquidsoap_custom.liq:
```diff
@@ -1,488 +1,512 @@
+    ({6w and 18h-19h58m}, once(playlist_rnr666)),
+    ({6w and 20h-21h58m}, once(playlist_mood_sequence)),
+    ({6w and 6w23h-7w00h28m}, once(playlist_d23)),
+    ({6w12h-7w00h}, playlist_off_air_ambient),                     #ambient: Merites - Gestalt
```
Yeah, the label was wrong. Fixed it. For the listening experience, it would probably better to stop playing gap-filling ambient after Mood, cause there is a one hour empty slot until D23 and it's a Saturday ;)

gammaw commented on azuracast/liquidsoap_custom.liq:
```diff
@@ -1,488 +1,512 @@
+    ({1w and 13h30m-23h30m}, playlist_off_air_ambient),                        #ambient: Who Say - Infinite
+    ({2w and 12h-13h58m}, once(playlist_lahmaloudclouds)),
+    ({2w and 15h30m-16h58m}, once(playlist_melyvagas)),
+    ({2w and 17h-18h58m}, once(playlist_radical_happiness)),
```
That's a bad typo. Thanks for finding it!

That's why it's good that we found a solution for an elegant handling of daylight saving time adjustments. I hope it will work smoothly.

gammaw commented on azuracast/liquidsoap_custom.liq:
```diff
@@ -1,488 +1,512 @@
+    ({2w and 20h-20h58m}, once(playlist_havizaj)),
+    ({2w and 21h-21h58m}, once(playlist_turmeric_acid)),
+    ({2w and 22h-22h58m}, once(playlist_exit)),
+    ({2w and 16h-22h30m}, playlist_off_air_ambient),                        #ambient: Melyvagas - Exit
```
Like for Lahma presents.

gammaw commented:
@tuz666 thanks for checking it and for the findings! I added a new commit w/ the fix, added more doc to the PR description, and merged formally master to this branch (practically an overwrite). Please review.
